ios/Classes/codec/AblyFlutterReader.m

```diff
@@ -5,15 +5,13 @@
 #import "AblyPlatformConstants.h"
 #import <ably_flutter/ably_flutter-Swift.h>
 
-static ARTLogLevel _logLevel(NSNumber *const number) {
-    switch (number.unsignedIntegerValue) {
-        case 99: return ARTLogLevelNone;
-        case 2: return ARTLogLevelVerbose;
-        case 3: return ARTLogLevelDebug;
-        case 4: return ARTLogLevelInfo;
-        case 5: return ARTLogLevelWarn;
-        case 6: return ARTLogLevelError;
-    }
+static ARTLogLevel _logLevel(NSString *const logLevelString) {
```

Comment:
Just a comment on this that doesn't need a code change and is just a recommendation for the future reference. it's common to use a dictionary in cases where keys / values are known at compile time. In such methods you would usually query this dictionary for the existence of a value and dealt with the case when the value does not exist.
Something like this:
```
if(logLevel[logLevelString]){
      return logLevel[logLevelString];
}
return ARTLogLevelWarn;
```

Reply:
You mentioned this previously and I have considered it.

- Using a map might be fewer characters, but it breaks up the code. Your code snippet has to check for existence in the map (developers have to check the map to double check when coding, and this is mental overhead). This an additional layer of indirection. What the code really is doing is a more implicit/hidden if-else anyway, it's clearer to see it happen IMHO. Also,
 - I could put in the method, at which point we are recreating the map on every method invocation. This might be bad (maybe not).
 - If I put the map outside the method, it is more mental overhead for a developer to jump to that map.
- It doesn't provide any additional checks or protection. The compiler warns us about the return types in both cases.
- It is not flexible. What if one day we want to deserialise a nested value. Using a map actually makes the code more rigid, and we are unable to call methods. In the enum case, this doesn't matter, but then we have 2 patterns in the codebase, as opposed to the shorter syntax I am using.

In the near future, I would like to rewrite this file in Swift, and perhaps split it up into separate files. At that time, I will implement both ways to have a better appreciation.

Reply:
I am resolving this conversation on @ikbalkaya's behalf.